pkg/backup/backup.go

```diff
@@ -501,8 +548,151 @@ func (kb *kubernetesBackupper) BackupWithResolvers(
+				log.WithFields(logrus.Fields{
+					"groupResource": relatedItem.GroupResource,
+					"namespace":     relatedItem.Namespace,
+					"name":          relatedItem.Name,
+				}).Warnf("Related item was not found in Kubernetes API, can't add to item block")
+				continue
+			}
+			if err != nil {
+				log.Error(errors.WithStack(err))
+				continue
+			}
+			itemsMap[relatedItem] = append(itemsMap[relatedItem], &kubernetesResource{
+				groupResource: relatedItem.GroupResource,
+				preferredGVR:  gvr,
+				namespace:     relatedItem.Namespace,
+				name:          relatedItem.Name,
+				inItemBlock:   true,
+			})
+			log.Infof("adding %s %s/%s to ItemBlock", relatedItem.GroupResource, relatedItem.Namespace, relatedItem.Name)
+			itemBlock.AddUnstructured(relatedItem.GroupResource, item, gvr)
+			kb.executeItemBlockActions(log, item, relatedItem.GroupResource, relatedItem.Name, relatedItem.Namespace, itemsMap, itemBlock)
+		}
+	}
+}
+
+func (kb *kubernetesBackupper) backupItemBlock(itemBlock BackupItemBlock) []schema.GroupResource {
+	// find pods in ItemBlock
+	// filter pods based on whether they still need to be backed up
+	// this list will be used to run pre/post hooks
+	var postHookPods []itemblock.ItemBlockItem
+	var items []itemblock.ItemBlockItem
+	itemBlock.Log.Debug("Executing pre hooks")
+	for _, item := range itemBlock.Items {
+		if item.Gr == kuberesource.Pods {
+			skipHooks := false
+			metadata, err := meta.Accessor(item.Item)
+			if err != nil {
+				itemBlock.Log.WithError(errors.WithStack(err)).Error("Error accessing pod metadata")
+				skipHooks = true
+			}
+			// Don't run hooks if pod is excluded
+			if !skipHooks && !itemBlock.itemBackupper.itemInclusionChecks(itemBlock.Log, false, metadata, item.Item, item.Gr) {
+				skipHooks = true
+			}
+			if !skipHooks {
+				key := itemKey{
+					resource:  resourceKey(item.Item),
+					namespace: metadata.GetNamespace(),
+					name:      metadata.GetName(),
+				}
+				// Don't run hooks if pod has already been backed up
+				if _, exists := itemBlock.itemBackupper.backupRequest.BackedUpItems[key]; !exists {
+					if err := itemBlock.itemBackupper.itemHookHandler.HandleHooks(itemBlock.Log, item.Gr, item.Item, itemBlock.itemBackupper.backupRequest.ResourceHooks, hook.PhasePre, itemBlock.itemBackupper.hookTracker); err != nil {
+						itemBlock.Log.WithError(err).WithField("name", item.Item.GetName()).Error("Error running hooks for pod")
+						// if pre hook fails, flag pod as backed-up and move on
+						itemBlock.itemBackupper.backupRequest.BackedUpItems[key] = struct{}{}
+						continue
+					}
+					postHookPods = append(postHookPods, item)
+				}
+			}
+		}
+		items = append(items, item)
+	}
+
+	itemBlock.Log.Debug("Backing up items in BackupItemBlock")
+	var grList []schema.GroupResource
+	for _, item := range items {
+		if backedUp := kb.backupItem(itemBlock.Log, item.Gr, itemBlock.itemBackupper, item.Item, item.PreferredGVR, &itemBlock); backedUp {
```

Comment:
For PVB, PVC is not to be backed up. How do we skip it since PVC and pod are in the same itemBlock?

Reply:
But we do back up PVC. Current workflow backs up the item. It's in the item list. So we don't  want to skip it here.

Reply:
Understood. As the current logics, we reply on the handler of the PVC to decide whether to skip the PVC or not. See [this comment](https://github.com/vmware-tanzu/velero/pull/8102#discussion_r1737646097) for the problems I can see.

Let's discuss in that comment.